### PR TITLE
Homework 2 and 3

### PR DESCRIPTION
**Pull request by olegkarman**

### CODE REVIEW

**Review by DMIbragimoov — APPROVED** [review]
LGTM

File: ProjectToTest/User.cs [hunk 1]
```diff
@@ -1,21 +1,31 @@
-﻿using System.Reflection;
-
-namespace ProjectToTest
+﻿namespace ProjectToTest
 {
+    internal enum UserGender
```
Comment (Oleksandr1823) [inline ProjectToTest/User.cs:3]: It is better to extract into a separate file

**Review by Oleksandr1823 — APPROVED** [review]

File: ProjectToTest/User.cs (hunk 1 above)
Comment (frostysh) [reply ProjectToTest/User.cs:3]: :O Enum into a separate file? But why?

**Review by frostysh — COMMENTED** [review]
Why You, Mr. Oleg changed the project from our last home work instead of just to make a new repository??? You just modified the "Program.cs" file from exercise about GitHub... xD

I will make my home exercise mineself, and then I will post Approve-request with the comments directly into the source code. I just do not want to 'spoil' mineself.

File: ProjectToTest/User.cs (hunk 1 above)
Comment (olegkarman) [reply ProjectToTest/User.cs:3]: It is a common practice to store different data in the different files, but this project is small and I don’t see a need in that.

**Comment by olegkarman** [comment]
@frostysh as it is my repository, I decided to take advantage in using it as I pleased. Furthermore, I don’t want to create a new test repo

File: ProjectToTest/Program.cs [hunk 2]
```diff
@@ -1,205 +1,71 @@
...
-
-            try
+            if (int.TryParse(Console.ReadLine(), out int genderInput) && Enum.IsDefined(typeof(UserGender), genderInput))
             {
-                Console.WriteLine("PLEASE, TYPE P NUMERIC INTEGER — COEFFICIENT:");
-                P = int.Parse(Console.ReadLine());
-
-                P_CHECK = CONST_0 / P; //CHECKS IF P = 0.
+                user1.Gender = (UserGender)genderInput;
+                break;
```
Comment (frostysh) [inline ProjectToTest/Program.cs:54]: Why we need `break` operator here? OMG — Loop there, I see now!

File: ProjectToTest/Program.cs [hunk 3]
```diff
@@ -1,205 +1,71 @@
-namespace HOMEWORK3_QUADRATIC_YPARKHOMENKO
+using ProjectToTest;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+internal class Program
 {
-    internal class Program
+    public static bool IsValidName(string name = "")
```
Comment (frostysh) [inline ProjectToTest/Program.cs:7]: A method with default value? Why??? Actually default value of string is `null` I suppose. Why to initialize it?

File: ProjectToTest/Program.cs [hunk 4]
```diff
@@ -1,205 +1,71 @@
...
-            Console.WriteLine("Hello from Nina!");      // HELLO FROM STUDENT NINA.
-            Console.WriteLine("Goodbye from Illia!");     // Goodbye FROM STUDENT ILLIA.
-            Console.WriteLine("For what its worth?");   // HELLO FROM STUDENT YAROSLAV.
-
-            Console.WriteLine($"\nTHIS PROGRAM SOLVING A QUADRATIC EQUATIONS OF THE FORM\n\nAx\u00B2 + Bx + C = 0,\n\nAND THE SECOND FORM\n\n(Px + Q)(Rx + S) = 0,\n\nA, P, B, Q, C, R, S — ARE COEFFITIENTS THE EACH OF THEM IS AN INTEGER NUMBER (MEANS NO FRACTIONAL PART) THAT CAN BE LOWER THAN 0, THE POSSIBLE RANGE OF INTEGER VALUES FOR A SINGLE VARIABLE ON ELECTRONIC CALCULATION MACHINE IS ({int.MinValue}, {int.MaxValue})\n\n");
-
-            byte CHECK_CONSOLE_AT_START = 0;
-            string SECOND_TYPE = "UNASSIGNED"; //ASSIGNMENT OF STRING VARIABLE THAT REQUIRED FOR OUR MECHANISM. TO READ IT IN "CATCH" PROGRAM.
+        Regex regex = new Regex(@"^[A-Za-z]+$");
+        return !string.IsNullOrWhiteSpace(name) && regex.IsMatch(name);
```
Comment (frostysh) [inline ProjectToTest/Program.cs:10]: IsMatch() is a static stuff — no need the instance to call it — (https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.ismatch?view=net-8.0), or mineself mistakes?

**Review by frostysh — APPROVED** [review]
Well, compared to others, it is good, I am approving. To be said, we have choose different approaches to setup properties in the instance. Me confused little bit with my own approach.

**Pull request merged**